implementation/src/main/java/io/smallrye/config/SmallRyeConfig.java

```diff
@@ -211,6 +230,31 @@ public String getRawValue(String name) {
         return getOptionalValue(name, Converters.newCollectionConverter(converter, collectionFactory));
     }
 
+    @Experimental("TODO")
+    public <T> T getConfigProperties(Class<T> klass) {
+        return getConfigProperties(klass, "");
+    }
+
+    @Experimental("TODO")
+    public <T> T getConfigProperties(Class<T> klass, String prefix) {
+        try {
+            final ConfigMappingProvider.Result result = configMappingProvider.mapConfiguration(this);
+            final T configRoot = result.getConfigRoot(prefix, klass);
+            // Build one on the fly
+            // Think if we want to merge the unknown one with the current configMapping.
```

Comment:
This I still don't agree with; it defeats the purpose of much of this API which was designed to uniformly validate a part of the configuration space.  Mapped configuration should be established while the configuration is being built IMO; after all, there is still a 1:1 mapping between "application" and configuration.

Reply:
Forgetting about this, what about the integration with the builder and having `ConfigMappingProvider` as part of `SmallRyeConfig` in here: https://github.com/smallrye/smallrye-config/blob/b46864f662f9434f66d67ab4b833902dcf24f81f/implementation/src/main/java/io/smallrye/config/SmallRyeConfig.java#L79

Is this something that you agree?

Reply:
I think it could make sense to be able to define the config interface mapping in the main builder, sure.

Reply:
Ok, I'll remove the "on the fly build" and keep the builder interface.

BTW, I didn't change the Mapping Entry API, so it can still be used like you proposed originally.